src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+        hw, hl, hh = self.hw, self.hl, self.hh
+        # Note: 2D show method assumes cyclic order of first 4 vertices
+        return (
+            self.relativePosition(Vector(self.hw,  self.hl,  self.hh)),
+            self.relativePosition(Vector(-self.hw, self.hl,  self.hh)),
+            self.relativePosition(Vector(-self.hw, -self.hl, self.hh)),
+            self.relativePosition(Vector(self.hw,  -self.hl, self.hh)),
+            self.relativePosition(Vector(self.hw,  self.hl,  -self.hh)),
+            self.relativePosition(Vector(-self.hw, self.hl,  -self.hh)),
+            self.relativePosition(Vector(-self.hw, -self.hl, -self.hh)),
+            self.relativePosition(Vector(self.hw,  -self.hl, -self.hh)),
+        )
+
+    @cached_property
+    def occupiedSpace(self):
+        """A region representing the space this object occupies"""
+        return MeshVolumeRegion(mesh=self.shape.mesh, \
+                dimensions=(self.width, self.length, self.height), \
+                position=self.position, rotation=self.orientation)
+
+    @cached_property
+    def boundingBox(self):
+        """A region representing this object's bounding box"""
+        return MeshVolumeRegion(self.occupiedSpace.mesh.bounding_box, center_mesh=False)
+
+    @cached_property
+    def inradius(self):
+        """A lower bound on the inradius of this object"""
+        # First check if all needed variables are defined. If so, we can
+        # compute the inradius exactly.
+        if not needsSampling([self.width, self.length, self.height, self.shape]):
+            shapeRegion = MeshVolumeRegion(mesh=self.shape.mesh, \
+                dimensions=(self.width, self.length, self.height))
+
+            return shapeRegion.inradius
+
+        # If we havea uniform distribution over shapes and a supportInterval for each dimension,
+        # we can compute a supportInterval for this object's inradius
+
+        # Define helper class
+        class InradiusHelper:
+            def __init__(self, support):
+                self.support = support
+
+            def supportInterval(self):
+                return self.support
+
+        # Extract bounds on all dimensions
+        min_width, max_width = supportInterval(self.width)
+        min_length, max_length = supportInterval(self.length)
+        min_height, max_height = supportInterval(self.height)
+
+        if any(val == None for val in [min_width, max_width, 
+                                       min_length, max_length,
+                                       min_height, max_height]):
+            # Can't get a bound on one or more dimensions, abort
+            return 0
+
+        min_bounds = np.array([min_width, min_length, min_height])
+        max_bounds = np.array([max_width, max_length, max_height])
+
+        # Extract a list of possible shapes
+        if isinstance(self.shape, Shape):
+            shapes = [self.shape]
+        elif hasattr(self.shape, "options"):
```

Comment:
This test isn't reliable. I would use `isinstance(self.shape, MultiplexerDistribution)`.